data-prepper-pipeline-parser/src/main/java/org/opensearch/dataprepper/pipeline/parser/transformer/DynamicConfigTransformer.java

```diff
@@ -72,6 +74,8 @@ public class DynamicConfigTransformer implements PipelineConfigurationTransforme
     private static final String JSON_PATH_IDENTIFIER = "$.";
     private static final String ARRAY_NODE_PATTERN = "([^\\[]+)\\[(\\d+)\\]$";
     private static final String SOURCE_COORDINATION_IDENTIFIER_ENVIRONMENT_VARIABLE = "SOURCE_COORDINATION_PIPELINE_IDENTIFIER";
+    private static final String SINK_SUBPIPELINE_PLUGIN_NAME = "pipeline";
+    private static final String SUBPIPELINE_PATH = "$.source.pipeline";
```

Comment:
Sub-pipelines aren't really a concept in Data Prepper. We can also give this a clearer name. e.g. `SOURCE_PIPELINE_PATH`.